### ISSUE TITLE
/usr/share/clearwater/bin/create_user creates undeletable users if cancelled

### ISSUE THREAD

**Issue by rkd-msw** [Metaswitch/crest#336]
#### Symptoms

 I ran `/usr/share/clearwater/bin/create_user 1500000..1600000 prov.clearwater.test password` to create 100,000 users. After doing so, I realised I wanted to run it with `time` to see how long it took, so I hit Ctrl-C and reran the command.

I then got this output:

```
clearwater@prov-homestead-1:~$ time 1500000..1600000 prov.clearwater.test password -k
ERROR: Private ID 1500000@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500001@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500002@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500003@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500004@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500005@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500006@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500007@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500008@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500009@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500010@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500011@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500012@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500013@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500014@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500015@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500016@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500017@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500018@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500019@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500020@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500021@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500022@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500023@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500024@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500025@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500026@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500027@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500028@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500029@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500030@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500031@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500032@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500033@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500034@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500035@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500036@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500037@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500038@prov.clearwater.test already exists - not creating
ERROR: Private ID 1500039@prov.clearwater.test already exists - not creating
ERROR: Failed to retrieve private IDs for public ID sip:1500039@prov.clearwater.test - HTTP status code 404
ERROR: Failed to retrieve iFC for public ID sip:1500039@prov.clearwater.test - HTTP status code 404
ERROR: Failed to find any information for public ID sip:1500039@prov.clearwater.test.
```

I tried to tidy up sip:1500039@prov.clearwater.test but delete_user failed:

```
clearwater@prov-homestead-1:~$ /usr/share/clearwater/bin/delete_user -f 1500039 prov.clearwater.test
Are you sure you wish to delete the following users: 1500039? [y/n]
y
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/share/clearwater/clearwater-prov-tools/env/lib/python2.7/site-packages/clearwater_prov_tools-0.1-py2.7.egg/metaswitch/ellis/prov_tools/delete_user.py", line 86, in <module>
    main()
  File "/usr/share/clearwater/clearwater-prov-tools/env/lib/python2.7/site-packages/clearwater_prov_tools-0.1-py2.7.egg/metaswitch/ellis/prov_tools/delete_user.py", line 77, in main
    if not utils.delete_user(private_id, public_id, force=args.force):
  File "/usr/share/clearwater/clearwater-prov-tools/env/local/lib/python2.7/site-packages/clearwater_prov_tools-0.1-py2.7.egg/metaswitch/ellis/prov_tools/utils.py", line 193, in delete_user
    homestead.delete_private_id(private_id, callback)
  File "/usr/share/clearwater/clearwater-prov-tools/env/local/lib/python2.7/site-packages/clearwater_prov_tools-0.1-py2.7.egg/metaswitch/ellis/remote/homestead.py", line 136, in delete_private_id
    irs = json.loads(associated_irs_response.body)['associated_implicit_registration_sets'][0]
IndexError: list index out of range
clearwater@prov-homestead-1:~$
```

#### Impact

After cancelling create_user, the user it was creating at that point is in an unrecoverable bad state.

#### Release and environment 

10.0.0-170127-~255.00.0-170127.020342

#### Steps to reproduce

Run  /usr/share/clearwater/bin/create_user and hit Ctrl-C. Run it again.

**Comment by rkd-msw** [comment]
The diags for https://github.com/Metaswitch/clearwater-cassandra/issues/126 are from this system and will have the relevant logs.

**Comment by sebrexmetaswitch** [comment]
Fix is merged, will be in the next nightly build. Back to you to re-test.

**Comment by rkd-msw** [comment]
Fixed in 10.0.0-170211-~255.00.0-170211.020437.
